element-connector/src/main/java/org/eclipse/californium/elements/tcp/TcpClientConnector.java

```diff
@@ -59,10 +67,11 @@ public TcpClientConnector(int numberOfThreads, int connectTimeoutMillis, int idl
 		this.numberOfThreads = numberOfThreads;
 		this.connectionIdleTimeoutSeconds = idleTimeout;
 		this.connectTimeoutMillis = connectTimeoutMillis;
-		this.listenUri = URI.create(String.format("%s://127.0.0.1:0", getSupportedScheme()));
+		this.listenUri = URI.create(String.format("%s://0.0.0.0:0", getSupportedScheme()));
```

Comment:
I think that is the whole point since this is a client connector only that initiates connections to servers only but cannot itself serve connections initiated by others.